lib/active_merchant/billing/gateways/pin.rb

```diff
@@ -36,18 +36,18 @@ def purchase(money, creditcard, options = {})
         commit(:post, 'charges', post, options)
       end
 
-      # Create a customer and associated credit card. The token that is returned
-      # can be used instead of a credit card parameter in the purchase method
-      def store(creditcard, options = {})
+      # Create a customer and associated payment method. The token that is returned
+      # can be used instead of a payment method parameter in the purchase method
+      def store(payment_method, options = {})
```

Comment:
This will not work, the [POST /customers](https://pinpayments.com/developers/api-reference/customers#post-customers) endpoint does not support customer creation with a `payment_source_token`, only embedded `card` parameters or a `card_token`, until this is supported you'd need to send the card token returned under `response.source.token` from the [POST /payment_sources](https://pinpayments.com/developers/api-reference/payment-sources#post-payment_sources) endpoint.

Reply:
Thanks, Andy. I'm going to create a ticket for our internal CI team to address this as well as some other issues I think I've identified. It looks like a lot of this integration is a bit old and may have experienced some technical drift, so it probably warrants some attention.